### PR TITLE
Implemented Lures and Repel/Lure "use another" menu

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## Description
- Implemented Lures by using a mask to reuse `VAR_REPEL_STEP_COUNT`.
  - Increases shiny chances. (Reworked shiny rerolls to support further additional rolls)
  - Increased chances of rarer Pokémon appearing by inverting the table encounter slot chances 20% of the time (credit CitrusBolt and Jaizu).
  - Increased encounter rate.
  - Encountered Wild Pokémon will have the highest possible level + 1.
- Implemented 4 configurations for the Repel/Lure prompt, togglable between them in `include/item_config.h`:
  - `I_REPEL_LURE_MENU` set to `FALSE`:
    - **1A** `VAR_LAST_REPEL_LURE_USED` not assigned: RSE to BW behavior.
    - **1B** `VAR_LAST_REPEL_LURE_USED` assigned: B2W2+ behavior.
  - `I_REPEL_LURE_MENU` set to `TRUE`:
    - **2A** `VAR_LAST_REPEL_LURE_USED` not assigned: Opens a menus that lets the player choose between the repel/lure types they have left. If they only have one, it defaults to behavior **1B**.
    - **2B**`VAR_LAST_REPEL_LURE_USED` assigned: Same as **2A** but the cursor for the menu automatically points to the last used item.

Repel menu based off @DizzyEggg's

## **Discord contact info**
AsparagusEduardo#6051

### CODE REVIEW

**Review by Jaizu — COMMENTED** [review]
Trailing spaces 👀
Edit: in wild_encounter.c

File: src/wild_encounter.c [hunk 1]
```diff
@@ -181,58 +184,82 @@ static void FeebasSeedRng(u16 seed)
...
-        return 11;
+        wildMonIndex = 11;
+
+    if (LURE_STEP_COUNT != 0 && (Random() % 10 < 2))
+        swap = TRUE;
+
+	if (swap)
+		wildMonIndex = 11 - wildMonIndex;
+
+	return wildMonIndex;
```
Comment (LOuroboros) [inline src/wild_encounter.c:222]:
```suggestion
    if (swap)
        wildMonIndex = 11 - wildMonIndex;

    return wildMonIndex;
```

File: src/wild_encounter.c [hunk 2]
```diff
@@ -181,58 +184,82 @@ static void FeebasSeedRng(u16 seed)
...
 }
 
 // ROCK_WILD_COUNT / WATER_WILD_COUNT
 static u8 ChooseWildMonIndex_WaterRock(void)
 {
+	u8 wildMonIndex = 0;
+	bool8 swap = FALSE;
```
Comment (LOuroboros) [inline src/wild_encounter.c:229]:
```suggestion
    u8 wildMonIndex = 0;
    bool8 swap = FALSE;
```

File: src/wild_encounter.c [hunk 3]
```diff
@@ -181,58 +184,82 @@ static void FeebasSeedRng(u16 seed)
...
-        return 4;
+        wildMonIndex = 4;
+
+    if (LURE_STEP_COUNT != 0 && (Random() % 10 < 2))
+        swap = TRUE;
+
+	if (swap)
+		wildMonIndex = 4 - wildMonIndex;
+
+	return wildMonIndex;
```
Comment (LOuroboros) [inline src/wild_encounter.c:249]:
```suggestion
    if (swap)
        wildMonIndex = 4 - wildMonIndex;

    return wildMonIndex;
```

File: src/wild_encounter.c [hunk 4]
```diff
@@ -181,58 +184,82 @@ static void FeebasSeedRng(u16 seed)
...
 }
 
 // FISH_WILD_COUNT
 static u8 ChooseWildMonIndex_Fishing(u8 rod)
 {
     u8 wildMonIndex = 0;
+	bool8 swap = FALSE;
```
Comment (LOuroboros) [inline src/wild_encounter.c:256]:
```suggestion
    bool8 swap = FALSE;
```

File: src/wild_encounter.c [hunk 5]
```diff
@@ -241,13 +268,19 @@ static u8 ChooseWildMonIndex_Fishing(u8 rod)
         else
             wildMonIndex = 1;
         break;
+
+		if (swap)
+			wildMonIndex = 1 - wildMonIndex;
```
Comment (LOuroboros) [inline src/wild_encounter.c:273]:
```suggestion
        if (swap)
            wildMonIndex = 1 - wildMonIndex;
```
Shouldn't this be put above the `break;`

File: src/wild_encounter.c [hunk 6]
```diff
@@ -241,13 +268,19 @@ static u8 ChooseWildMonIndex_Fishing(u8 rod)
...
     case GOOD_ROD:
         if (rand < ENCOUNTER_CHANCE_FISHING_MONS_GOOD_ROD_SLOT_2)
             wildMonIndex = 2;
         if (rand >= ENCOUNTER_CHANCE_FISHING_MONS_GOOD_ROD_SLOT_2 && rand < ENCOUNTER_CHANCE_FISHING_MONS_GOOD_ROD_SLOT_3)
             wildMonIndex = 3;
         if (rand >= ENCOUNTER_CHANCE_FISHING_MONS_GOOD_ROD_SLOT_3 && rand < ENCOUNTER_CHANCE_FISHING_MONS_GOOD_ROD_SLOT_4)
             wildMonIndex = 4;
+
+		if (swap)
+			wildMonIndex = 6 - wildMonIndex;
```
Comment (LOuroboros) [inline src/wild_encounter.c:283]:
```suggestion
        if (swap)
            wildMonIndex = 6 - wildMonIndex;
```

File: src/wild_encounter.c [hunk 7]
```diff
@@ -260,46 +293,61 @@ static u8 ChooseWildMonIndex_Fishing(u8 rod)
             wildMonIndex = 8;
         if (rand >= ENCOUNTER_CHANCE_FISHING_MONS_SUPER_ROD_SLOT_8 && rand < ENCOUNTER_CHANCE_FISHING_MONS_SUPER_ROD_SLOT_9)
             wildMonIndex = 9;
+
+		if (swap)
+			wildMonIndex = 14 - wildMonIndex;
```
Comment (LOuroboros) [inline src/wild_encounter.c:298]:
```suggestion
        if (swap)
            wildMonIndex = 14 - wildMonIndex;
```

**Review by LOuroboros — CHANGES_REQUESTED** [review]

File: src/wild_encounter.c [hunk 8]
```diff
@@ -241,13 +268,19 @@ static u8 ChooseWildMonIndex_Fishing(u8 rod)
         else
             wildMonIndex = 1;
         break;
+
+        if (swap)
+            wildMonIndex = 1 - wildMonIndex;
```
Comment (LOuroboros) [inline src/wild_encounter.c:273]: I guess I forgot to press the button to send this on my review, but shouldn't this `if (swap)` check be put *before* the `break;`?

Comment (AsparagusEduardo) [reply src/wild_encounter.c:273]: ...yes.

Comment (Jaizu) [reply src/wild_encounter.c:273]: Yes it should. That's a mistake on my end.

Comment (AsparagusEduardo) [reply src/wild_encounter.c]: Fixed now.

**Pull request merged**